stratify/tests/test_bounded_vinterp.py

```diff
@@ -0,0 +1,199 @@
+    # [broadcasting_dims, axis_interpolation, z_varying].
+    # This this will take the form:
+    # [broadcasting_dims, axis_interpolation, 1].
+    def setUp(self):
+        self.data = np.ones((4, 6))
+        self.bounds = self.gen_bounds(0, 7, 1)
+
+    def gen_bounds(self, start, stop, step):
+        bounds = np.vstack([np.arange(start, stop-step, step),
+                            np.arange(start+step, stop, step)])
+        bounds = bounds.transpose((1, 0))
+        return bounds.copy()
+
+    def test_target_half_resolution(self):
+        target_bounds = self.gen_bounds(0, 7, 2)
+        res = bounded_vinterp.interpolate_conservative(target_bounds,
+                                                       self.bounds, self.data,
+                                                       axis=1)
+        target_data = np.ones((4, 3))*2
+        assert_array_equal(res, target_data)
+
+    def test_target_double_resolution(self):
+        target_bounds = self.gen_bounds(0, 6.5, 0.5)
+        res = bounded_vinterp.interpolate_conservative(target_bounds,
+                                                       self.bounds, self.data,
+                                                       axis=1)
+        target_data = np.ones((4, 12))*.5
+        assert_array_equal(res, target_data)
+
+    def test_no_broadcasting(self):
+        # In the case of no broadcasting, transposed and reshaped array of form
+        # [broadcasting_dims, axis_interpolation, z_varying] should end up
+        # [1, axis_interpolation, 1].
+        data = self.data[0]
+        target_bounds = self.gen_bounds(0, 7, 2)
+        res = bounded_vinterp.interpolate_conservative(target_bounds,
+                                                       self.bounds, data,
+                                                       axis=0)
+        target_data = np.ones((3))*2
+        assert_array_equal(res, target_data)
+
+    def test_target_extends_above_source(self):
+        # |-|-|-|-|-|-|-|   - Source
+        # |-|-|-|-|-|-|-|-| - Target
+        source_bounds = self.gen_bounds(0, 7, 1)
+        target_bounds = self.gen_bounds(0, 8, 1)
+        res = bounded_vinterp.interpolate_conservative(target_bounds,
+                                                       source_bounds,
+                                                       self.data, axis=1)
+        target_data = np.ones((4, 7))
+        target_data[:, -1] = 0
+        assert_array_equal(res, target_data)
+
+    def test_target_extends_below_source(self):
+        #   |-|-|-|-|-|-|-|   - Source
+        # |-|-|-|-|-|-|-|-|   - Target
+        source_bounds = self.gen_bounds(0, 7, 1)
+        target_bounds = self.gen_bounds(-1, 7, 1)
+        res = bounded_vinterp.interpolate_conservative(target_bounds,
+                                                       source_bounds,
+                                                       self.data, axis=1)
+        target_data = np.ones((4, 7))
+        target_data[:, 0] = 0
```

Comment:
Why 0 and not some other extrapolation scheme?

Reply:
Hmm, I think you have a point.  Feels to me like 'nan' (as you say) is more appropriate here.  If no source points are contributing then there is no data so it should be 'nan'.

I can't think why I chose 0 in the first place and since @hdyson didn't pick up on this either, @hdyson is there a reason that you can remember why 0 seemed to make sense at the time but doesn't now? :)

Cheers

Reply:
That's going back a while.  From memory (so pinch of salt and all that), I seem to recall we discussed this on the phone, but I don't remember the reasons for and against 0.  

That said, this behaviour is consistent with the approach we implemented for the CMIP6 aerosols (and was in turn copied from an earlier IDL routine), so it could simply be for consistency with legacy applications.  Hope that helps?

Reply:
Thanks @hdyson I'll change to 'NaN' then, we can always post-process the returned array if it proves to be necessary to the UM from such ancillaries.